pymongo/cursor.py

```diff
@@ -73,7 +73,6 @@
 _CURSOR_CLOSED_ERRORS = frozenset(
     [
         43,  # CursorNotFound
-        50,  # MaxTimeMSExpired
```

Comment:
If MaxTimeMSExpired happens that means a timeout has expired and the subsequent killCursors will either 1) timeout immediately (eg pymongo.timeout) or 2) run but exceed the user's timeout (eg maxTimeMS). Perhaps we should call `self.__die(False)` in the case of timeouts to defer the killCursors to the background thread.

Reply:
Are there other timeout errors not in this list we should do the same behavior for?

Reply:
Good question. Yes, you can check the PyMongoError.timeout property.